clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThreadTest.java

```diff
@@ -211,178 +219,52 @@ public void testResetPositionsProcessFailureIsIgnored() {
-        String groupId = "group-id";
-        NetworkClientDelegate.UnsentRequest request = new NetworkClientDelegate.UnsentRequest(
-            new FindCoordinatorRequest.Builder(
-                new FindCoordinatorRequestData()
-                    .setKeyType(FindCoordinatorRequest.CoordinatorType.TRANSACTION.id())
-                    .setKey(groupId)),
-            Optional.empty());
-
-        networkClient.add(request);
-        assertTrue(networkClient.hasAnyPendingRequests());
-        assertFalse(networkClient.unsentRequests().isEmpty());
-        assertFalse(client.hasInFlightRequests());
+    public void testSendUnsentRequests() {
+        when(networkClientDelegate.hasAnyPendingRequests()).thenReturn(true).thenReturn(true).thenReturn(false);
         consumerNetworkThread.cleanup();
-
-        assertTrue(networkClient.unsentRequests().isEmpty());
-        assertFalse(client.hasInFlightRequests());
-        assertFalse(networkClient.hasAnyPendingRequests());
-    }
-
-    private void prepareOffsetCommitRequest(final Map<TopicPartition, Long> expectedOffsets,
-                                            final Errors error,
-                                            final boolean disconnected) {
-        Map<TopicPartition, Errors> errors = partitionErrors(expectedOffsets.keySet(), error);
-        client.prepareResponse(offsetCommitRequestMatcher(expectedOffsets), offsetCommitResponse(errors), disconnected);
-    }
-
-    private Map<TopicPartition, Errors> partitionErrors(final Collection<TopicPartition> partitions,
-                                                        final Errors error) {
-        final Map<TopicPartition, Errors> errors = new HashMap<>();
-        for (TopicPartition partition : partitions) {
-            errors.put(partition, error);
-        }
-        return errors;
-    }
-
-    private OffsetCommitResponse offsetCommitResponse(final Map<TopicPartition, Errors> responseData) {
-        return new OffsetCommitResponse(responseData);
-    }
-
-    private MockClient.RequestMatcher offsetCommitRequestMatcher(final Map<TopicPartition, Long> expectedOffsets) {
-        return body -> {
-            OffsetCommitRequest req = (OffsetCommitRequest) body;
-            Map<TopicPartition, Long> offsets = req.offsets();
-            if (offsets.size() != expectedOffsets.size())
-                return false;
-
-            for (Map.Entry<TopicPartition, Long> expectedOffset : expectedOffsets.entrySet()) {
-                if (!offsets.containsKey(expectedOffset.getKey())) {
-                    return false;
-                } else {
-                    Long actualOffset = offsets.get(expectedOffset.getKey());
-                    if (!actualOffset.equals(expectedOffset.getValue())) {
-                        return false;
-                    }
-                }
-            }
-            return true;
-        };
+        assertFalse(networkClientDelegate.hasAnyPendingRequests());
```

Comment:
I'm afraid we lost some changes here? we were verifying the networkClientDelegate.poll 2 times (and wanted to remove the assert on the mock)

Reply:
Whoops, yeah I highlighted the wrong lines by accident. Fixed that now